### PR TITLE
feat: adding ability for for charts to be pulled with plain HTTP

### PR DESCRIPTION
**Pull request by ennekein**
accomplished by:

- Adding PlainHttp attribute to RepositorySpec, HelmDefault, ReleaseSpec
- When set to true, pass `--plain-http` to helm commands

Tweaked https://github.com/helmfile/helmfile/pull/1228 by @hoangelos

Resolves https://github.com/helmfile/helmfile/discussions/1224

### CODE REVIEW

**Comment by ennekein** [comment]
@yxxhero I see the test stage has failed, but I do not see any failed test. Any idea what the issue could be?

**Comment by yxxhero** [comment]
@ennekein oh. you can find the failed case by searching `failed:`. `ctrl+F`

**Comment by yxxhero** [comment]
@ennekein sorry. update the search keyword `fail:`.

**Comment by ennekein** [comment]
@yxxhero it gives me no result.

However, I ran the tests in my VS Code and saw that `TestGenerateID` is failing. Do we need to update the hashes manually when we change the contents of `ReleaseSpec`?

**Comment by yxxhero** [comment]
@ennekein yeah. you should update.

File: pkg/state/state.go [hunk 1]
```diff
@@ -221,6 +223,7 @@ type RepositorySpec struct {
 	Keyring         string `yaml:"keyring,omitempty"`
 	PassCredentials bool   `yaml:"passCredentials,omitempty"`
 	SkipTLSVerify   bool   `yaml:"skipTLSVerify,omitempty"`
+	PlainHttp       bool   `yaml:"plainHttp,omitempty"`
```
Comment (yxxhero) [inline pkg/state/state.go:226]: where this field be used?

**Comment by yxxhero** [comment]
@ennekein oh. BTW. please fix DCO. follow this link.

File: pkg/state/state.go (hunk 1 above)
Comment (ennekein) [reply pkg/state/state.go:226]: It is used here: https://github.com/helmfile/helmfile/pull/1672/files/bab7cd3092d9d25cb2a4fb476a81d9a9737ae3fd#diff-7e4e87a1e777423a3d9a915e00fb78a9c4fc2f4c476863f60b67e89b9c635697R2864-R2866

Which is used to pull charts from OCI repositories in `getOCIChart` in `PrepareCharts`.

**Comment by ennekein** [comment]
Should be ready for review now

File: pkg/state/state.go [hunk 2]
```diff
@@ -2597,13 +2603,21 @@ func (st *HelmState) kubeConnectionFlags(release *ReleaseSpec) []string {
...
+	var repoSkipTLSVerify, repoPlainHttp bool
+	repo, _ := st.GetRepositoryAndNameFromChartName(release.Chart)
+	if repo != nil {
+		repoPlainHttp = repo.PlainHttp
+		repoSkipTLSVerify = repo.SkipTLSVerify
+	}
+
+	if release.PlainHttp || st.HelmDefaults.PlainHttp || repoPlainHttp {
+		flags = append(flags, "--plain-http")
+	} else if release.InsecureSkipTLSVerify || st.HelmDefaults.InsecureSkipTLSVerify || repoSkipTLSVerify {
```
Comment (yxxhero) [inline pkg/state/state.go:2616]: please don't use `else if`? maybe we can return directly?

**Comment by yxxhero** [comment]
@ennekein LGTM. did you test it locally?

**Comment by ennekein** [comment]
@yxxhero I did build it locally and tested it with both HTTP and HTTPS registries.

**Review by yxxhero — APPROVED** [review]

**Pull request merged**